### PR TITLE
chore(DmaBuf): Add a helper to dump contents of DmaBuf

### PR DESCRIPTION
**Pull request by dsharma-dc**
Helper utility for debugging the contents of dma buffer used in io-engine. This dumps the buffer data in traces.

### CODE REVIEW

**Comment by dsharma-dc** [comment]
@tiagolobocastro @dsavitskiy I was using this to debug some workflow. If you think this can be helpful in general, we can merge this.
Looks like this in traces, if using `buf.dump()`:

```
[2023-10-13T04:48:41.419120717+00:00  INFO spdk_rs::dma:dma.rs:107] DmaBuf(IoVec(iovec { iov_base: 0x200011e0df00, iov_len: 65 }))
[:00:00:00:00:01:2c:00:00:00:00:00:00:00:31:30:2e
:31:2e:30:2e:36:3a:38:34:32:30:2f:6e:71:6e:2e:32
:30:31:39:2d:30:35:2e:69:6f:2e:6f:70:65:6e:65:62
:73:3a:72:65:70:6c:31:6e:31:00:00:00:05:00:00:00
:00]
```

**Review by dsavitskiy — APPROVED** [review]

File: src/dma.rs [hunk 1]
```diff
@@ -90,6 +90,22 @@ impl DmaBuf {
     pub fn to_io_vec(&self) -> IoVec {
         self.0
     }
+
+    /// Helper to dump content of a DmaBuf.
+    /// Derived Debug trait for DmaBuf only prints base address and length.
+    pub fn dump(&self) {
```
Comment (tiagolobocastro) [inline src/dma.rs]: Should this return a String and allow the caller to use whatever print kind it wants?

**Review by tiagolobocastro — APPROVED** [review]

File: src/dma.rs (hunk 1 above)
Comment (dsharma-dc) [reply src/dma.rs]: Yeah, that can be done.

**Comment by dsharma-dc** [comment]
bors try

**Pull request merged**